synchrad/calc.py

```diff
@@ -527,16 +528,29 @@ def _init_comm(self):
             self.ocl_version = "None"
         else:
             try:
-                self.ctx = cl.create_some_context(**ctx_kw_args)
+                print(f"Creating context with args: {ctx_kw_args}")  # Logging
+
+                # Set up OpenCL context
+                platforms = cl.get_platforms()
+                gpus = platforms[0].get_devices(device_type=cl.device_type.GPU)
+
+                # Map MPI rank to a specific GPU
+                device = gpus[self.rank % len(gpus)]
+                self.ctx = cl.Context(devices=[device])
```

Comment:
This is the new context selection logic, I tested it and it works on a multi-GPU system, but feel free to fine-tune it.

Reply:
Can we mark this as resolved or do you want to change anything here?